symengine/lib/symengine_wrapper.pyx

```diff
@@ -2078,22 +2081,134 @@ class DenseMatrixIter(object):
 
     next = __next__
 
-Matrix = DenseMatrix
+cdef class MutableDenseMatrix(DenseMatrixBase):
 
-cdef matrix_to_vec(DenseMatrix d, symengine.vec_basic& v):
+    def row_join(self, rhs):
+        cdef DenseMatrixBase o = _sympify(rhs)
+        if self.rows != o.rows:
+            raise ShapeError("`self` and `rhs` must have the same number of rows.")
+        cdef DenseMatrixBase result = zeros(self.rows, self.cols + o.cols)
+        for i in range(self.rows):
+            for j in range(self.cols):
+                result[i, j] = self[i, j]
+        for i in range(o.rows):
+            for j in range(o.cols):
+                result[i, j + self.cols] = o[i, j]
+        return result
+
+    def col_join(self, bott):
+        cdef DenseMatrixBase o = _sympify(bott)
+        if self.cols != o.cols:
+            raise ShapeError("`self` and `rhs` must have the same number of columns.")
+        cdef DenseMatrixBase result = zeros(self.rows + o.rows, self.cols)
+        for i in range(self.rows):
+            for j in range(self.cols):
+                result[i, j] = self[i, j]
+        for i in range(o.rows):
+            for j in range(o.cols):
+                result[i + self.rows, j] = o[i, j]
+        return result
+
+    def col_del(self, i):
+        if i < -self.cols or i >= self.cols:
+            raise IndexError("Index out of range: 'i=%s', valid -%s <= i < %s"
+                             % (i, self.cols, self.cols))
+        cdef DenseMatrixBase result = zeros(self.rows, self.cols -1)
+        for j in range(self.rows):
+            for k in range(self.cols - 1):
+                if k >= i:
+                    result[j, k] = self[j, k + 1]
+                else:
+                    result[j, k] = self[j, k]
+        return result
+
+    def col_op(self, j, f):
+        raise NotImplementedError
+
+    def col_swap(self, i, j):
+        for k in range(0, self.rows):
+            self[k, i], self[k, j] = self[k, j], self[k, i]
+
+    def copyin_list(self, key, value):
+        raise NotImplementedError
+
+    def copyin_matrix(self, key, value):
```

Comment:
I implemented as much functionality as I could from `SymPy` so that later when I'm porting changes, I don't get issues in cases where the function is called directly in `SymPy`, as I'd like to make minimal changes to `SymPy`.